packages/devextreme/js/__internal/ui/chat/chat.ts

```diff
@@ -21,13 +33,46 @@ class Chat extends Widget<Properties> {
     $(this.element()).addClass(CHAT_CLASS);
 
     super._initMarkup();
+
+    this._renderHeader();
+    this._renderMessageList();
+    this._renderMessageBox();
+  }
+
+  _renderHeader(): void {
+    const { title } = this.option();
+
+    this._chatHeader = this._createComponent($('<div>'), ChatHeader, { title });
+
+    $(this._chatHeader.element()).appendTo(this.element());
+  }
+
+  _renderMessageList(): void {
+    const { items } = this.option();
+
+    this._messageList = this._createComponent($('<div>'), MessageList, {
```

Comment:
It is bad practice to create a component on an element that is not attached to the DOM. First, we add the element to the page, and then we create the component on this element. This helps avoid many rendering issues.

Reply:
> This helps avoid many rendering issues

Like what? It is better to create all elements first and then render once, than to run reflow repaint processes every time some parameters are updated. It's better for efficiency

Reply:
Rendering Issues: Components may not render correctly or as expected if they are not attached to the DOM. This can lead to incorrect layouts or visual artifacts. Many our components require calculations for proper rendering. For example, a text area is unlikely to be rendered correctly using the maxHeight option, or a floating label may not be positioned correctly.
You can get more details from our support team; they can provide information about scenarios where components are rendered on elements that are detached from the DOM